packages/create-emotion-styled/src/index.js

```diff
@@ -85,7 +88,9 @@ function createEmotionStyled(emotion: Emotion, view: ReactType) {
           }
         }
         render() {
-          const { props, state } = this
+          const state = this.state
+          const props = Object.assign({}, defaultProps, this.props)
```

Comment:
just a thought - it generally should be the same, but maybe worth to offload that work to builtin React's mechanism of `defaultProps`? u could just assign `static defaultProps = options.props`

Reply:
https://github.com/emotion-js/emotion/pull/617/files#diff-77b4dbda613d78e500ded003e8362f64R1125

Reply:
Maybe you are onto something though. Maybe there is a need for `defaultProps` and `withProps`

Reply:
@tkh44 I don't think it would fail, the example below would fail with defaultProps though and probably does with this current implementation as well

```jsx
const SomeComponent = styled('div', { props: { className:'thing' }})(...styles)

render(<SomeComponent className="other-thing" />)
```